### PR TITLE
Hdc1008

### PR DESCRIPTION
**Pull request by iceaway**
## Summary
Example application showing how to use the hdc1008 driver (see pull request 975: https://github.com/apache/incubator-nuttx/pull/975). Let me know if the copyright stuff needs to change.

### CODE REVIEW

**Comment by xiaoxiang781216** [comment]
Should we remove the 2nd patch and merge the 3rd into the 1st? also the copyright need change to Apache for the new file.

File: examples/hdc1008_demo/Make.defs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+# OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+# AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
```
Comment (patacongo) [inline examples/hdc1008_demo/Make.defs:35]: This must have an Apache 2.0 header.  Use the one from nuttx/sched/sched/sched_getcpu.c.  Note that there is no author or copyright in that header.

File: examples/hdc1008_demo/Makefile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+# OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+# AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+############################################################################
```
Comment (patacongo) [inline examples/hdc1008_demo/Makefile:34]: This must have an Apache 2.0 header.  Use the one from nuttx/sched/sched/sched_getcpu.c.  Note that there is no author or copyright in that header.

File: examples/hdc1008_demo/hdc1008_main.c [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
```
Comment (patacongo) [inline examples/hdc1008_demo/hdc1008_main.c:34]: This must have an Apache 2.0 header.  Use the one from nuttx/sched/sched/sched_getcpu.c.  Note that there is no author or copyright in that header.

**Comment by iceaway** [comment]
Headers are now updated to the apache 2.0 license. Not sure what you meant about merging the patches @xiaoxiang781216 . Is that something I can do on my branch or is it done by whoever merges this to master?

**Comment by xiaoxiang781216** [comment]
> Headers are now updated to the apache 2.0 license. Not sure what you meant about merging the patches @xiaoxiang781216 . Is that something I can do on my branch or is it done by whoever merges this to master?

This PR has four patch, but it make sense to merge into one like this:
git rebase --interactive HEAD~4
select squach
git push -f origin hdc1008

**Comment by iceaway** [comment]
I think I screwed something up now....

**Comment by patacongo** [comment]
Try rebasing against the master then doing a force push to your branch.  That might do it.

**Comment by xiaoxiang781216** [comment]
If something get wrong, create a new branch, cherry-pick and rebase the change again.

**Comment by iceaway** [comment]
Go ahead and close this one, I opened up a new pull request.

**Pull request closed without merging**